### PR TITLE
[iOS] Fix issue creating rectangle using Radius with a high value

### LINKED ISSUE #11033: [Bug] iOS Native crash when RadiusX/RadiusY > Width/Height of Shapes.Rectangle
### Steps to Reproduce

```xaml

xmlns:shapes="clr-namespace:Xamarin.Forms.Shapes;assembly=Xamarin.Forms.Core"
....

 <shapes:Rectangle 
                              WidthRequest="100"
                              HeightRequest="100"
                              RadiusX="200"
                              RadiusY="200"
                              StrokeLineCap="Round"
                              StrokeThickness="2"
                              StrokeLineJoin="Round"
                              Stroke="Red"
                              Rotation="45"/>
```

### Expected Behavior

I don't know. A circle?

### Actual Behavior

```
	  at <unknown> <0xffffffff>
	  at CoreGraphics.CGPath:CGPathAddRoundedRect <0x00157>
	  at CoreGraphics.CGPath:AddRoundedRect <0x000e2>
	  at Xamarin.Forms.Platform.iOS.RectView:UpdateShape <0x00212>
	  at Xamarin.Forms.Platform.iOS.RectView:UpdateRadiusX <0x00152>
	  at Xamarin.Forms.Platform.iOS.RectangleRenderer:UpdateRadiusX <0x0018a>
	  at Xamarin.Forms.Platform.iOS.RectangleRenderer:OnElementChanged <0x001da>
	  at Xamarin.Forms.Platform.iOS.VisualElementRenderer`1:SetElement <0x00adf>
	  at Xamarin.Forms.Platform.iOS.VisualElementRenderer`1:Xamarin.Forms.Platform.iOS.IVisualElementRenderer.SetElement <0x000c2>
	  at Xamarin.Forms.Platform.iOS.Platform:CreateRenderer <0x00174>
	  at Xamarin.Forms.Platform.iOS.VisualElementPackager:OnChildAdded <0x0028a>
	  at Xamarin.Forms.Platform.iOS.VisualElementPackager:Load <0x001a6>
	  at Xamarin.Forms.Platform.iOS.VisualElementRenderer`1:SetElement <0x0086a>
	  at Xamarin.Forms.Platform.iOS.VisualElementRenderer`1:Xamarin.Forms.Platform.iOS.IVisualElementRenderer.SetElement <0x000c2>
	  at Xamarin.Forms.Platform.iOS.Platform:CreateRenderer <0x00174>
	  at Xamarin.Forms.Platform.iOS.VisualElementPackager:OnChildAdded <0x0028a>
	  at Xamarin.Forms.Platform.iOS.VisualElementPackager:OnChildAdded <0x00120>
	  at Xamarin.Forms.Element:OnChildAdded <0x0017b>
	  at Xamarin.Forms.VisualElement:OnChildAdded <0x00092>
	  at Xamarin.Forms.Page:OnInternalAdded <0x00140>
	  at Xamarin.Forms.Page:InternalChildrenOnCollectionChanged <0x0068a>
	  at System.Collections.ObjectModel.ObservableCollection`1:OnCollectionChanged <0x000fa>
	  at System.Collections.ObjectModel.ObservableCollection`1:OnCollectionChanged <0x00103>
	  at System.Collections.ObjectModel.ObservableCollection`1:InsertItem <0x00132>
	  at System.Collections.ObjectModel.Collection`1:Add <0x0019c>
	  at Xamarin.Forms.TemplateUtilities:OnContentChanged <0x0033b>
	  at Xamarin.Forms.BindableObject:SetValueActual <0x007c7>
	  at Xamarin.Forms.BindableObject:SetValueCore <0x00c62>
	  at Xamarin.Forms.BindableObject:SetValue <0x003f2>
	  at Xamarin.Forms.BindableObject:SetValue <0x0008a>
	  at Xamarin.Forms.Xaml.ApplyPropertiesVisitor:TrySetValue <0x009ba>
	  at Xamarin.Forms.Xaml.ApplyPropertiesVisitor:SetPropertyValue <0x00882>
	  at Xamarin.Forms.Xaml.ApplyPropertiesVisitor:Visit <0x01fe2>
	  at Xamarin.Forms.Xaml.ElementNode:Accept <0x007fc>
	  at Xamarin.Forms.Xaml.RootNode:Accept <0x00602>
	  at Xamarin.Forms.Xaml.XamlLoader:Visit <0x0070e>
	  at Xamarin.Forms.Xaml.XamlLoader:Load <0x00ae2>
	  at Xamarin.Forms.Xaml.XamlLoader:Load <0x0008a>
	  at Xamarin.Forms.Xaml.XamlLoader:Load <0x002a2>
	  at Xamarin.Forms.Xaml.Extensions:LoadFromXaml <0x00082>
	  at RoundyAndTheShapeMachine.MainPage:__InitComponentRuntime <0x00062>
	  at RoundyAndTheShapeMachine.MainPage:InitializeComponent <0x002f2>
	  at System.Object:runtime_invoke_void__this__ <0x00195>
	  at <unknown> <0xffffffff>
	  at System.Reflection.RuntimeMethodInfo:InternalInvoke <0x00100>
	  at System.Reflection.RuntimeMethodInfo:Invoke <0x0055a>
	  at System.Reflection.MethodBase:Invoke <0x00087>
	  at <<ReloadXamlOnMainThreadAsync>b__0>d:MoveNext <0x0069a>
	  at MoveNextRunner:InvokeMoveNext <0x000ab>
	  at System.Threading.ExecutionContext:RunInternal <0x00585>
	  at System.Threading.ExecutionContext:Run <0x00092>
	  at MoveNextRunner:Run <0x001aa>
	  at System.Threading.Tasks.AwaitTaskContinuation:InvokeAction <0x00093>
	  at System.Threading.Tasks.AwaitTaskContinuation:RunCallback <0x00153>
	  at System.Threading.Tasks.SynchronizationContextAwaitTaskContinuation:Run <0x0013a>
	  at System.Threading.Tasks.Task:FinishContinuations <0x006a6>
	  at System.Threading.Tasks.Task:FinishStageThree <0x0015a>
	  at System.Threading.Tasks.Task`1:TrySetResult <0x0022a>
	  at System
.Threading.Tasks.TaskCompletionSource`1:TrySetResult <0x00092>
	  at System.Threading.Tasks.TaskCompletionSource`1:SetResult <0x0007a>
	  at <>c__DisplayClass1_0:<AnimateTo>b__1 <0x0008a>
	  at <>c__DisplayClass14_0`1:<AnimateInternal>b__1 <0x000c9>
	  at Xamarin.Forms.AnimationExtensions:HandleTweenerFinished <0x00690>
	  at Xamarin.Forms.Tweener:<Start>b__27_0 <0x004d6>
	  at Xamarin.Forms.Internals.Ticker:SendSignals <0x0031e>
	  at Xamarin.Forms.Internals.Ticker:SendSignals <0x00102>
	  at Xamarin.Forms.Platform.iOS.CADisplayLinkTicker:<EnableTimer>b__7_0 <0x0007a>
	  at Foundation.NSActionDispatcher:Apply <0x0006f>
	  at System.Object:runtime_invoke_void__this__ <0x000e5>
	  at <unknown> <0xffffffff>
	  at UIKit.UIApplication:UIApplicationMain <0x00254>
	  at UIKit.UIApplication:Main <0x000b2>
	  at UIKit.UIApplication:Main <0x00132>
	  at RoundyAndTheShapeMachine.iOS.Application:Main <0x00092>
	  at <Module>:runtime_invoke_void_object <0x00342>
===========================================================
```

### Basic Information

- Version with issue: 4.7.0-pre4
- Last known good version: n/a
- IDE: 16.7.0 P2
- Platform Target Frameworks: 
  - iOS:  13.5

### Reproduction Link

[RoundyAndTheShapeMachine.zip](https://github.com/xamarin/Xamarin.Forms/files/4772643/RoundyAndTheShapeMachine.zip)

### Workaround

Don't do that.

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Fix issue creating rectangle using Radius with a high value on iOS.

### Issues Resolved ### 

- fixes #11033

### API Changes ###
 
 None

### Platforms Affected ### 

- iOS

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

fix11033

### Testing Procedure ###
Launch Core Gallery and navigate to the issue 11033. If the Ellipse renders without problems, the tests has passed.

### PR Checklist ###
- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Platform.iOS/Shapes/RectangleRenderer.cs [hunk 1]
```diff
@@ -11,6 +11,8 @@ namespace Xamarin.Forms.Platform.MacOS
 {
     public class RectangleRenderer : ShapeRenderer<Rect, RectView>
     {
+        const double MaximumRadius = 0.5d;
```
Comment (rmarinho) [inline Xamarin.Forms.Platform.iOS/Shapes/RectangleRenderer.cs:14]: Why this value?

Comment (jsuarezruiz) [reply Xamarin.Forms.Platform.iOS/Shapes/RectangleRenderer.cs:15]: The maximum value for the cornerHeight and cornerWidth using `AddRoundedRect` is the middle of the Height and Width.
https://developer.apple.com/documentation/coregraphics/cgmutablepath/2427144-addroundedrect

Added a comment in the code.

**Review by rmarinho — APPROVED** [review]

File: Xamarin.Forms.Platform.iOS/Shapes/RectangleRenderer.cs [hunk 2]
```diff
@@ -47,14 +50,22 @@ void UpdateRadius()
...
+            Control.UpdateRadiusX(radiusX);
         }
 
         void UpdateRadiusY()
         {
-            if (Element.Height > 0)
-                Control.UpdateRadiusY(Element.RadiusY / Element.Height);
+            var radiusY = ValidateRadius(Element.RadiusY / Element.HeightRequest);
+            Control.UpdateRadiusY(radiusY);
+        }
```
[the remark this answers is not in the document]
Comment (mattleibow) [reply Xamarin.Forms.Platform.iOS/Shapes/RectangleRenderer.cs:61]: This does result in a different outcome on Android - they still use `Width`/`Height`.

[the remark this answers is not in the document]
Comment (mattleibow) [reply Xamarin.Forms.Platform.iOS/Shapes/RectangleRenderer.cs:61]: I see Tizen uses the requests. Testing UWP now.

**Review by mattleibow — CHANGES_REQUESTED** [review]
This is a breaking change, but also leaves the others inconsistent.

File: Xamarin.Forms.Platform.iOS/Shapes/RectangleRenderer.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (jsuarezruiz) [reply Xamarin.Forms.Platform.iOS/Shapes/RectangleRenderer.cs]: Reviewed and now `HeightRequest` and WidthRequest is used everywhere to fix the inconsistent issue.

[the remark this answers is not in the document]
Comment (mattleibow) [reply Xamarin.Forms.Platform.iOS/Shapes/RectangleRenderer.cs]: @jsuarezruiz should we be using Width (actual layout size) or WidthRequest (what the user said, but could be anything and will usually be different when set to Expand?)

File: Xamarin.Forms.Platform.iOS/Shapes/RectangleRenderer.cs (hunk 1 above)
Comment (mattleibow) [reply Xamarin.Forms.Platform.iOS/Shapes/RectangleRenderer.cs:15]: Love the comment!

**Review by mattleibow — APPROVED** [review]

**Pull request merged**